src/repositories/catalogue.repository.js

```diff
@@ -0,0 +1,49 @@
+import prisma from '../config/db.config.js'
+
+const catalogueBody = {
+  catalogue_id: true,
+  type: true,
+  code: true,
+  description: true
+}
+
+export const getAllCatalogsRepository = async () => {
+  return prisma.catalogue.findMany({
+    select: catalogueBody
+  })
+}
+
+export const getSingleCatalogueRepository = async id => {
+  return prisma.catalogue.findUnique({
+    where: {
+      catalogue_id: id
+    },
+    select: catalogueBody
+  })
+}
+
+export const createCatalogueRepository = async data => {
+  return prisma.catalogue.create({
+    data,
+    select: catalogueBody
+  })
+}
+
+export const updateCatalogueRepository = async ({ data, id }) => {
```

Comment:
Why do you decompose the object that you compose inLine when calling this function?
```javascript
const catalogueUpdated = await updateCatalogueRepository({ data, id: Number(id) })
```